x/interchainstaking/keeper/receipt_test.go

```diff
@@ -314,3 +317,58 @@ func (suite *KeeperTestSuite) TestReceiptStore() {
 
 	suite.Equal(&now, receipt.Completed)
 }
+
+func (suite *KeeperTestSuite) TestSendTokenIBC() {
+	suite.Run("test", func() {
+		suite.SetupTest()
+
+		// setup transfer channel
+		suite.path.EndpointA.ChannelConfig.Version = transfertypes.Version
+		suite.path.EndpointB.ChannelConfig.Version = transfertypes.Version
+		suite.coordinator.CreateTransferChannels(suite.path)
+
+		suite.setupTestZones()
+
+		quicksilver := suite.GetQuicksilverApp(suite.chainA)
+		ctx := suite.chainA.GetContext()
+
+		zone, found := quicksilver.InterchainstakingKeeper.GetZone(ctx, suite.chainB.ChainID)
+		suite.True(found)
+
+		sender := suite.chainA.SenderAccount.GetAddress()
+		receiver := addressutils.GenerateAddressForTestWithPrefix("cosmos")
+
+		amount := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))
+		err := quicksilver.BankKeeper.MintCoins(ctx, types.ModuleName, sdk.NewCoins(amount))
+		suite.NoError(err)
+		err = quicksilver.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, sender, sdk.NewCoins(amount))
+		suite.NoError(err)
+
+		// Try to send the native token
+		err = quicksilver.InterchainstakingKeeper.SendTokenIBC(ctx, sender, receiver, &zone, amount)
+		suite.NoError(err)
+
+		portID := types.TransferPort
+		channelID := suite.path.EndpointA.ChannelID
+		// channelBID := suite.path.EndpointB.ChannelID
+
+		ibcAmount := transfertypes.GetTransferCoin(portID, channelID, sdk.DefaultBondDenom, sdk.NewInt(100))
+
+		err = quicksilver.BankKeeper.MintCoins(ctx, types.ModuleName, sdk.NewCoins(ibcAmount))
+		suite.NoError(err)
+		err = quicksilver.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, sender, sdk.NewCoins(ibcAmount))
+		suite.NoError(err)
+
+		quicksilver.TransferKeeper.SetDenomTrace(
+			ctx,
+			transfertypes.DenomTrace{
+				Path:      fmt.Sprintf("%s/%s", portID, channelID),
+				BaseDenom: sdk.DefaultBondDenom,
+			},
+		)
+
+		// Try to send the ibc token
+		err = quicksilver.InterchainstakingKeeper.SendTokenIBC(ctx, sender, receiver, &zone, ibcAmount)
```

Comment:
I think we should check the change in the receiver's balance after `SendTokenIBC` success, and add one more test case for [this](https://github.com/quicksilver-zone/quicksilver/blob/a24dc7029618e046d66b2af2028925cca7ba6338/x/interchainstaking/keeper/receipt.go#L140)

Reply:
There is no way to check the balance after testcase, refered to [TestSendTransfer](https://github.com/cosmos/ibc-go/blob/830982adfb4448573f8898bbd07a73d06bff640a/modules/apps/transfer/keeper/relay_test.go#L21) (ibc-go test)

Reply:
Ok, just need to add one more test case then we can merge it

Reply:
@ThanhNhann added, please check

Reply:
Is it worth checking the balance of the escrow module as a check?

```
amount := quicksilver.TransferKeeper.GetTotalEscrowForDenom(suite.chainA.GetContext(), coin.GetDenom())
suite.Require().Equal(expEscrowAmount, amount.Amount)
```

Reply:
@phamminh0811 did the suggestion above not work?

Reply:
there is no GetTotalEscrowForDenom function in transferkeeper

Reply:
Can you check [this](https://github.com/cosmos/ibc-go/blob/3d9f652d11cece3dbd9aa9c204d7ef6b43b111b3/modules/apps/transfer/keeper/keeper.go#L220) @phamminh0811

Reply:
@ThanhNhann I think it is on ibc-go v8. So right now we don't have that function yet